app/screens/browse_channels/channel_dropdown.tsx

```diff
@@ -0,0 +1,112 @@
+import {View, Text} from 'react-native';
+
+import CompassIcon from '@components/compass_icon';
+import {ITEM_HEIGHT} from '@components/slide_up_panel_item';
+import {useTheme} from '@context/theme';
+import {TITLE_HEIGHT} from '@screens/bottom_sheet/content';
+import {bottomSheet} from '@screens/navigation';
+import {makeStyleSheetFromTheme} from '@utils/theme';
+import {typography} from '@utils/typography';
+
+import {PUBLIC, SHARED} from './browse_channels';
+import DropdownSlideup from './dropdown_slideup';
+
+type Props = {
+    typeOfChannels: string;
+    onPress: (channelType: string) => void;
+    canShowArchivedChannels: boolean;
+    sharedChannelsEnabled: boolean;
+}
+
+const getStyleFromTheme = makeStyleSheetFromTheme((theme: Theme) => {
+    return {
+        channelDropdown: {
+            ...typography('Body', 100, 'SemiBold'),
+            lineHeight: 20,
+            color: theme.centerChannelColor,
+            marginLeft: 20,
+            marginTop: 12,
+            marginBottom: 4,
+        },
+        channelDropdownIcon: {
+            color: theme.centerChannelColor,
+        },
+    };
+});
+
+const BOTTOM_SHEET_HEIGHT_BASE = 55; // Magic number
+
+export default function ChannelDropdown({
+    typeOfChannels,
+    onPress,
+    canShowArchivedChannels,
+    sharedChannelsEnabled,
+}: Props) {
+    const intl = useIntl();
+    const theme = useTheme();
+    const style = getStyleFromTheme(theme);
+
+    // Depends on all props, so no need to use a callback.
+    const handleDropdownClick = () => {
+        const renderContent = () => {
+            return (
+                <DropdownSlideup
+                    canShowArchivedChannels={canShowArchivedChannels}
+                    onPress={onPress}
+                    sharedChannelsEnabled={sharedChannelsEnabled}
+                    selected={typeOfChannels}
+                />
+            );
+        };
+
+        let items = 1;
+        if (canShowArchivedChannels) {
+            items += 1;
+        }
+        if (sharedChannelsEnabled) {
+            items += 1;
+        }
+
+        bottomSheet({
+            title: intl.formatMessage({id: 'browse_channels.dropdownTitle', defaultMessage: 'Show'}),
+            renderContent,
+            snapPoints: [(items * ITEM_HEIGHT) + TITLE_HEIGHT + BOTTOM_SHEET_HEIGHT_BASE, 10],
+            closeButtonId: 'close',
+            theme,
+        });
+    };
+
+    let channelDropdownText;
+    if (typeOfChannels === PUBLIC) {
+        channelDropdownText = intl.formatMessage({id: 'browse_channels.showPublicChannels', defaultMessage: 'Show: Public Channels'});
+    } else if (typeOfChannels === SHARED) {
+        channelDropdownText = intl.formatMessage({id: 'browse_channels.showSharedChannels', defaultMessage: 'Show: Shared Channels'});
+    } else {
+        channelDropdownText = intl.formatMessage({id: 'browse_channels.showArchivedChannels', defaultMessage: 'Show: Archived Channels'});
+    }
```

Comment:
0/5 you could use the else fallback as the predefined value.

```suggestion
    let channelDropdownText = channelDropdownText = intl.formatMessage({id: 'browse_channels.showArchivedChannels', defaultMessage: 'Show: Archived Channels'});
    if (typeOfChannels === PUBLIC) {
        channelDropdownText = intl.formatMessage({id: 'browse_channels.showPublicChannels', defaultMessage: 'Show: Public Channels'});
    } else if (typeOfChannels === SHARED) {
        channelDropdownText = intl.formatMessage({id: 'browse_channels.showSharedChannels', defaultMessage: 'Show: Shared Channels'});
    }
```